kart/point_cloud/import_.py

```diff
@@ -0,0 +1,147 @@
+from glob import glob
+import json
+from os.path import expanduser
+import sys
+
+import click
+
+from kart.exceptions import (
+    InvalidOperation,
+    NotFound,
+    NO_IMPORT_SOURCE,
+    INVALID_FILE_FORMAT,
+)
+from kart.output_util import format_wkt_for_output
+
+
+@click.command("point-cloud-import", hidden=True)
+@click.pass_context
+@click.argument("sources", metavar="SOURCES", nargs=-1, required=True)
+def point_cloud_import(ctx, sources):
+    """
+    Experimental command for importing point cloud datasets. Work-in-progress.
+    Will eventually be merged with the main `import` command.
+
+    SOURCES should be one or more LAZ or LAS files (or wildcards that match multiple LAZ or LAS files).
+    """
+    import pdal
+
+    all_source_paths = []
+    for source in sources:
+        source_paths = list(glob(expanduser(source)))
+        if not source_paths:
+            raise NotFound(NO_IMPORT_SOURCE, f"No data found at {source}")
+        for source_path in source_paths:
+            all_source_paths.append(source_path)
+
+    version_set = ListBasedSet()
+    copc_version_set = ListBasedSet()
+    pdrf_set = ListBasedSet()
+    crs_set = ListBasedSet()
+
+    for source_path in all_source_paths:
+        click.echo(f"Checking {source_path}...          \r", nl=False)
```

Comment:
why the carriage return here?

Reply:
It prints all the "Checking {source_path}" one after the other on the same line, which might be good if you have hundreds of them. 
This could change a bit if I start processing them in parallel